### PR TITLE
Fix bucket resamplers trying to print non-existent name

### LINKED ISSUE #1647: Bucket resamplers AttributeError in logging message
The log message in the Bucket resamplers compute methods contains `data.name` which results in an AttributeError because `data` at this point is not an xarray Dataarray but a numpy array.

https://github.com/pytroll/satpy/blob/9257378761d6ac6c826719c33455dd1dee9fc7c9/satpy/resample.py#L1173

### PR DESCRIPTION
**Pull request by BENR0**
This fixes bucket resampler AttributeError from log message in compute methods.

 - [x] Closes #1647 
 - [x] Tests added

### CODE REVIEW

**Comment by djhoese** [comment]
Do you have an example piece of code that shows you running in to this error? As far as I can tell Satpy should always be passing a `DataArray` to `.compute`. :thinking: unless the resampler in pyresample does something different.

**Comment by BENR0** [comment]
I just use `scn.resample(area_def, resampler="bucket_avg")`. In Line 1112 the np.array is extracted from the DataArray and then handed to the `compute` method of the resampler:

https://github.com/pytroll/satpy/blob/9257378761d6ac6c826719c33455dd1dee9fc7c9/satpy/resample.py#L1112

At least for the `bucket_avg` resampler the bug was introduced in #1539 I think.

**Comment by djhoese** [comment]
@pnuu do you know if the compute method is supposed to be expecting a dask array or a DataArray? The referenced pull request adds the log message by @ameraner but the DataArray -> dask change was there before.

**Comment by pnuu** [comment]
According to the docstring `resampler.get_average()` expects to receive either dask or Numpy array. I think @BENR0 is correct and the resamplers are always passed a plain dask array.

**Comment by djhoese** [comment]
Ok so dask arrays should have a `.name` if I remember correctly, but I don't think it is useful as it is usually based on the last task in the dask graph and includes a generated UUID most of the time (not super human-readable). If we are trying to get the name of the DataArray then we should really be doing `data_arr.attrs['name']`...but if we are trying to print out the DataArray's identifier then we should really be printing out its DataID. @mraspaud is the proper way to get the DataID for a DataArray to check for `.attrs['_satpy_id']`?

**Comment by mraspaud** [comment]
Yes

File: satpy/resample.py [hunk 1]
```diff
@@ -1117,6 +1117,7 @@ def resample(self, data, **kwargs):
             dims = ('y', 'x')
         else:
             dims = data.dims
+        LOG.debug("Resampling %s", str(data.name))
```
Comment (djhoese) [inline satpy/resample.py:1120]:
```suggestion
        LOG.debug("Resampling %s", str(data.attrs.get('_satpy_id', 'unknown')))
```

**Pull request merged**